src/Blockcore/Networks/IPAddressExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Blockcore.Networks
+{
+    public static class IPAddressExtensions
+    {
+        public static IPAddress MapToIPv6(this IPAddress addr)
+        {
+            if (addr.AddressFamily != AddressFamily.InterNetwork)
+                throw new ArgumentException("Must pass an IPv4 address to MapToIPv6");
+
+            string ipv4str = addr.ToString();
+
+            return IPAddress.Parse("::ffff:" + ipv4str);
+        }
+
+        public static bool IsIPv4MappedToIPv6(this IPAddress addr)
```

Comment:
I am pretty sure the `IPAddress` of system ns has already that method

Reply:
@miladsoft @maftooh  this is the method I mean

https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.isipv4mappedtoipv6?view=net-6.0